galvasr2/dump_cc_by_licenses.py

```diff
@@ -1,3 +1,42 @@
 #!/usr/bin/env python
 
-print("TODO")
+from align.spark.schemas import ARCHIVE_ORG_SCHEMA
+from pyspark.sql import SparkSession
+
+spark = SparkSession.builder.appName('CC-BY-license').getOrCreate()
+
+def myConcat(*cols):
+    concat_columns = []
+    for c in cols[:-1]:
+        concat_columns.append(F.coalesce(c, F.lit("*")))
+        concat_columns.append(F.lit(" "))  
+    concat_columns.append(F.coalesce(cols[-1], F.lit("*")))
+    return F.concat(*concat_columns)
+
+def create_dump_license(input_catalogue_path:str="gs://the-peoples-speech-west-europe/archive_org/Mar_7_2021/EXPANDED_LICENSES_FILTERED_ACCESS.jsonl.gz", save_as:str='csv', spark=spark):
+    df = spark.read.format('json').schema(ARCHIVE_ORG_SCHEMA).load(input_catalogue_path)
+    ##Filter by necessary columns 
+    columns = [df.metadata.licenseurl, df.metadata.creator, df.metadata.title, df.metadata.credits]
+    df = df.select(columns)
+    ##Rename columns
+    df = (df.withColumnRenamed('metadata.licenseurl','licenseurl').withColumnRenamed('metadata.creator', 'creator')
+         .withColumnRenamed('metadata.title', 'title').withColumnRenamed('metadata.credits', 'credits'))
+    ##There only 4 register without license at the moment. Without information in the rest of the data
+    df = df.dropna(subset=['licenseurl'])
+    ##Regex filter to search any kind of "by" license
+    regexp = r'\b(http|https)://creativecommons.org/licenses/by\b'
```

Comment:
Something is not right with this regex.

Try out this code to see what I mean:

```
from pyspark.sql.types import StructType,StructField, StringType
from pyspark import Row
from pyspark.sql import SparkSession

schema = StructType([       
    StructField('licenseurl', StringType(), True),
])

spark = SparkSession.builder.appName('tester').getOrCreate()

regexp = r'\b(http|https)://creativecommons.org/licenses/by\b'
# My proposed new regex (fixes failing assertion below)
# regexp = r'(http|https)://creativecommons.org/licenses/by/(1[.]0|2[.]0|2[.]5|3[.]0|4[.]0)'

df = spark.createDataFrame(data=[("https://creativecommons.org/licenses/by-nc-nd/3.0/",), 
                                 ("https://creativecommons.org/licenses/by-sa/3.0/",),
                                 ("https://creativecommons.org/licenses/by/3.0/",),
                                 ("https://creativecommons.org/licenses/by/3.0",),
                                 ("https://creativecommons.org/licenses/by/3.0/it",),
                                 ("https://creativecommons.org/licenses/by/3.0/it/",),
                                ], 
                           schema=schema)
filtered_rows = df.filter(df.licenseurl.rlike(regexp)).collect()
assert filtered_rows == [Row(licenseurl="https://creativecommons.org/licenses/by/3.0/"),
                         Row(licenseurl="https://creativecommons.org/licenses/by/3.0"),
                         Row(licenseurl="https://creativecommons.org/licenses/by/3.0/it"),
                         Row(licenseurl="https://creativecommons.org/licenses/by/3.0/it/"),
                         ]
```

Basically, the trouble is that it accepts by-sa, by-nc, by-nc-nd, and all other sorts of licenses that we are not interested in. Now, we shouldn't be downloading any data with those licenses *anyway*, but an extra level of checking is probably good.

Reply:
@Ciroye can you respond to this comment before we merge?

Reply:
Let's stick with my suggested regex. The motivation is that I want to be as specific as possible. I would like to check for only license types that we already tried to explicitly download here: https://github.com/mlcommons/peoples-speech/blob/a50625276acc291643bc64ef5c4af0cdbfcfd0d9/scripts/archive.org/download_items.py#L8 (You would be right to say that we are repeating ourselves, but I'm not sure how to easily convert the apache lucene query language to a regex easily.)

My concern is legal issues. As you pointed out elsewhere, we have four works with NULL licenseurl fields, even though our archive.org search query should not match any works with NULL licenseurl fields. So it is possible that some works with inappropriate licenses may be output by the download script. This is an extra bit of redundancy, essentially.